SupportBank/HelperFunctions.cs

```diff
@@ -33,15 +38,33 @@ public void ListAccount(string name)
             }
         }
 
-        public void ParseFileAndPopulateDictionary()
+        public void ParseFileAndPopulateDictionary(string filename)
         {
-            using var reader = new StreamReader("./Transactions2014.csv");
+            logger.Info($"Reading file {filename}");
+            using var reader = new StreamReader($"./{filename}");
             using var csv = new CsvReader(reader, CultureInfo.InvariantCulture);
             csv.Configuration.PrepareHeaderForMatch = (string header, int index) => header.ToLower();
+            var options = new TypeConverterOptions
+            {
+                DateTimeStyle = DateTimeStyles.None,
+                Formats = new[] {"dd/MM/yyyy"},
+            };
+            csv.Configuration.TypeConverterOptionsCache.AddOptions<DateTime>(options);
+            logger.Info("File read successfully - Beginning to add transactions");
             var records = csv.GetRecords<Transaction>();
-            foreach (var transaction in records)
+            try
+            {
+                foreach (var transaction in records)
+                {
+                    AddTransaction(transaction);
+                }
+                logger.Info("Transactions added with no issues");
+            }
+            catch (TypeConverterException e)
             {
-                AddTransaction(transaction);
+                var message = $"Error on row {e.ReadingContext.Row} {e.Text} is not of correct data type in {filename}";
+                logger.Fatal(message);
+                Console.WriteLine(message + "\nRest of file has been ignored, please fix errors and run again. Or continue using with part of data missing\n");
```

Comment:
You get setup the logger so that it will also output to the console, that way you don't need to log twice 🙂

Reply:
Do you mean to set it up that way? Or that it's already set up that way? I didn't really want the console cluttered with lots of log messages so only decided to output this specific one as it was useful to the user

Reply:
I mean you can set it up that way, by adding a second target. You can set the minimum log level for each target, so if you don't want to clutter the console you can set the minimum log level to for example error 🙂